src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -349,12 +355,146 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
         trigger=TriggerUpdateParameters(
             source_triggers=source_trigger_update_params,
             base_image_trigger=base_image_trigger_update_params
+        ),
+        credentials=get_custom_registry_credentials(
+            cmd,
+            auth_mode=auth_mode
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_credential_add(cmd,
+                            client,
+                            task_name,
+                            registry_name,
+                            login_server=None,
+                            username=None,
+                            password=None,
+                            resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    if not login_server or not username or not password:
+        raise CLIError("--login-server, --username and --password are required to add a credential")
+
+    TaskUpdateParameters, PlatformUpdateParameters, AgentProperties, TriggerUpdateParameters = cmd.get_models(
+        'TaskUpdateParameters', 'PlatformUpdateParameters', 'AgentProperties', 'TriggerUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        status=None,
+        platform=PlatformUpdateParameters(
+            os=None
+        ),
+        agent_configuration=AgentProperties(
+            cpu=None
+        ),
+        timeout=None,
+        step=None,
+        trigger=TriggerUpdateParameters(
+            source_triggers=None,
+            base_image_trigger=None
+        ),
+        credentials=get_custom_registry_credentials(
+            cmd,
+            login_server=login_server,
+            username=username,
+            password=password
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_credential_update(cmd,
```

Comment:
How is this different from `acr_task_credential_add`?

Reply:
+1 to @djyou  comment, credential add can be add or update , from the server, if the same credential is passed , it will be updated.